src/NUnitFramework/framework/Attributes/RandomAttribute.cs

```diff
@@ -176,7 +178,7 @@ public IEnumerable GetData(IParameterInfo parameter)
                     _source = new SByteDataSource(_count);
                 else if (parmType == typeof(decimal))
                     _source = new DecimalDataSource(_count);
-                else if (parmType.IsEnum)
+                else if (parmType.GetTypeInfo().IsEnum)
```

Comment:
I used the name TypeInfo in several places in my Type wrapper. Maybe we should change that (some time) to avoid confusion with the "real" TypeInfo.